repos/system_upgrade/el7toel8/actors/systemfacts/libraries/systemfacts.py

```diff
@@ -206,23 +207,17 @@ def _parse(r):
 
 def get_selinux_status():
     ''' Get SELinux status information '''
-    def asbool(x):
-        return x == 'enabled'
-
-    def identity(x):
-        return x
-
-    key_mapping = (('SELinux status', 'enabled', asbool),
-                   ('Policy MLS status', 'mls_enabled', asbool),
-                   ('Loaded policy name', 'policy', identity),
-                   ('Current mode', 'runtime_mode', identity),
-                   ('Mode from config file', 'static_mode', identity),)
 
     outdata = {}
-    for item in call(['sestatus']):
-        for (key, mapped, cast) in key_mapping:
-            if item.startswith(key):
-                outdata[mapped] = cast(item[len(key)+1:].strip())
+    outdata['enabled'] = True if selinux.is_selinux_enabled() == 1 else False
+    outdata['mls_enabled'] = True if selinux.is_selinux_mls_enabled == 1 else False
+    outdata['runtime_mode'] = "enforcing" if selinux.security_getenforce() == 1 else "permissive"
+    enforce_mode = selinux.selinux_getenforcemode()
+    if enforce_mode >= 0:
+        outdata['static_mode'] = "enforcing" if enforce_mode == 1 else "permissive"
+    else:
+        outdata['static_mode'] = "disabled"
+    outdata['policy'] = selinux.selinux_getpolicytype()[1]
```

Comment:
regarding the thing we cannot set dependency in the SPEC file, we are not sure the library is installed.  From that point on systems without the `libselinux-python` RPM the script will fail unhandled. Move import here and use `try` and catch `ImportError` to inform user that the rpm has to be installed. additionally, put commented out requires in the SPEC file so we will know we need them but it will not be used.

```
try:
    import selinux
    outdata...
except ImportError:
    self.report_error(...)
```

Reply:
See https://github.com/oamg/leapp/pull/405 - to be able to use report_error in the library

Reply:
@pirat89 , copy.